### PR TITLE
Add Devices Resource

### PR DESCRIPTION
**Pull request by rounak-adhikary**
# Description
Add ome_devices resource

# ISSUE TYPE
- Feature Pull Request

##### RESOURCE OR DATASOURCE NAME
ome_devices resource

##### OUTPUT

image

image

```paste below
coverage: 91.4% of statements
ok      terraform-provider-ome/clients  66.446s coverage: 91.4% of statements
```

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

### CODE REVIEW

**Comment by terraform-providers-svc** [comment]
Can one of the admins verify this patch?

File: models/devices.go [hunk 1]
```diff
@@ -183,6 +212,96 @@ type OmeManagementProfileData struct {
...
+	}
+	var plan types.List
+	vals := strings.Split(id, ":")
+	var did, didst string
+	if l := len(vals); l == 1 {
+		did = "id"
+		didst = id
+	} else if l > 2 {
+		dgs.AddError(
+			"Wrong id format.",
```
Comment (Krishnan-Priyanshu) [inline models/devices.go:257]: ID

File: examples/resources/ome_devices/resource.tf [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  devices = [
+    {
+      service_tag = "ABCD34"
+    },
+    {
+      id = 2000
+    },
+    {
+      service_tag = "QWX321"
+    }
```
Comment (Krishnan-Priyanshu) [inline examples/resources/ome_devices/resource.tf:29]: we may add an example and tests for management_ips attribute.

File: models/devices.go (hunk 1 above)
Comment (rounak-adhikary) [reply models/devices.go:257]: You mean the "id" string in the diagnostics? Is there any particular reason for this request?
The reason I am asking is that, even though Go conventionally uses "ID" in most cases, the user facing side of terraform almost always goes for "id". This is seen in the required "id" attribute of  all resources as well as in the import block.

Comment (Krishnan-Priyanshu) [reply models/devices.go:257]: shall we mention this inside double quotes?
`Wrong \"id\" format.`

File: ome/resource_devices.go [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource              = &resourceDevices{}
+	_ resource.ResourceWithConfigure = &resourceDevices{}
+)
+
+// NewDevicesResource is new resource for application csr
```
Comment (shenda1) [inline ome/resource_devices.go:41]: application csr?

Comment (rounak-adhikary) [reply ome/resource_devices.go]: fixed

**Review by Krishnan-Priyanshu — APPROVED** [review]

**Pull request merged**